### PR TITLE
Feat/add callback for upload

### PR DESCRIPTION
**Pull request by PenghaiZhang**
This PR's focus is adding the support of calling a callback when an upload is completed. However, there are more commits included to resolve warnings and configuration issues. They are: 

1. Using URL rather than local package to install the Kaltura TS Client API;
2. Not generating type definition files for tests and storybooks;
3. Removing the use of `@typescript-eslint/recommended-requiring-type-checking`;
4. Resolving Rollup build warnings about `unresolved dependencies`;
5. Resolving warnings about bundling the whole Kaltura TS Client API (not perfectly addressed as you still can see the warning in test)

All npm tasks are passed after above changes.

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -18,4 +18,5 @@ export default {
     },
   ],
   plugins: [typescript()],
+  external: [ "react", "react-dom", "react-error-boundary", "kaltura-typescript-client", "kaltura-typescript-client/api/types", "lodash" ]
```
Comment (edalex-ian) [inline rollup.config.js:21]: So no joy with wildcards? (e.g. `"kaltura-typescript-client*"` or `"kaltura-typescript-client/**/*"`)

File: src/KalturaUploader.tsx [hunk 2]
```diff
@@ -94,12 +99,19 @@ const KalturaUploaderInternal = ({
...
+  callback,
 }: KalturaUploaderProps): JSX.Element => {
   const [state, dispatch] = useReducer(reducer, { id: "start" });
   const [mediaType, setMediaType] = useState<MediaTypeOptions>(
     MediaTypes.video
   );
 
+  useEffect(() => {
+    if (state.id === "complete") {
+      callback([state.entry]);
```
Comment (edalex-ian) [inline src/KalturaUploader.tsx:111]: Hmmm, interesting that the ESLint React Hooks plugin did not complain about this usage of `state.entry`. :thinking: 

Ah! We don't have that plugin setup yet. I'll do that in another branch. But you might want to add the dependency here.

**Review by edalex-ian — COMMENTED** [review]
Good work @PenghaiZhang , thanks for all the effort wrangling the build ecosystem too.

Just one query and then an effect dependency to add, but also: How about modifying `example/index.js` to have a callback function which is something like:

```js
const callback = (entries) => {
  console.log("Callback was called with the following entries:");
  console.log(entries);
  document.getElementById("root").textContent = "Process complete, see console log!";
};
```

File: rollup.config.js (hunk 1 above)
Comment (PenghaiZhang) [reply rollup.config.js:21]: Unfortunately wildcards don't work :cry:

**Review by edalex-ian — APPROVED** [review]
LGTM! Go ahead and merge, tag and release.

**Pull request merged**